Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -1007,4 +1007,43 @@ theorem sigma_eq_iff_eq_comp_cast {α : Type*} {a b : Σii, Fin ii → α} :
     sigma_eq_of_eq_comp_cast _ h'⟩
 #align fin.sigma_eq_iff_eq_comp_cast Fin.sigma_eq_iff_eq_comp_cast
 
+section ExtendFun
+
+/-- Extend a function `f : Fin n → α` to a function on all natural numbers, by
+    defining `f i = a` for all `i ≥ n` and some given constant `a` -/
+def extendFun {α : Type*} {n: ℕ} (f : Fin n → α) (a : α) : ℕ → α :=
+  fun i =>
+    if h : i < n then
+      f ⟨i, h⟩
+    else
+      a
```

Comment:
This is just
```suggestion
def extendFun {α : Type*} {n: ℕ} (f : Fin n → α) (a : α) : ℕ → α :=
  Function.extend Fin.val f n
```
so in fact I don't think we should add a new definition.

Reply:
Unfortunately `Function.extend` is noncomputable, so I don't think that's going to work for @alexkeizer. Perhaps we can use `Set.piecewise` instead?

Of course, I'd prefer we work out what makes the `Fin` version of the function so hard to use in #5920

Reply:
I've generalize `Set.piecewise` to 
```lean
def Set.piecewiseMem {α : Type u} {β : α → Sort v} (s : Set α) [DecidablePred (· ∈ s)]
    (f : ∀ i ∈ s, β i) (g : ∀ i ∉ s, β i) : ∀ i, β i :=
  fun i ↦ if h : i ∈ s then f i h  else g i h
```

And used this to define `Fin.extendFun`.

Reply:
I fixed the build, so all is green now.

Reply:
This has now been implemented by Alex.